src/typechecker.ts

```diff
@@ -179,6 +185,37 @@ const getExpressionVisitor =
     }
   };
 
+type ErrorForUsingTypeXAsTypeY =
+  | null
+  | 'number-to-math'
```

Comment:
For completion's sake, shouldn't we have `bigint-to-real`?

Reply:
Sure I guess. That's not related to this PR but I can fold it in.